bin/stringsCompletenessTest.py

```diff
@@ -0,0 +1,140 @@
+    pattern = re.compile('_getString\(\"(.*)\"\)|mozL10n\.get\(\"(.*)\"\)')
+
+    # iterate search through files in code directories
+    props_found = 0
+    for code_dir in def_code_dirs:
+        os.chdir(loop_root_dir)
+        print("Current directory: ", os.getcwd())
+        print("code_dir", code_dir)
+        os.chdir(code_dir)
+        print("Current directory: ", os.getcwd())
+        for fname in multiple_file_types("*.js", "*.jsx", "*.jsm"):
+            file_path = os.path.join(os.getcwd(), fname)
+            print(fname)
+            print("fname", fname)
+            print("file_path", file_path)
+            with io.open(file_path, "r", encoding="UTF-8") as code_file:
+                for line in code_file:  # process line by line
+                    # print(line)
+                    for match in re.finditer(pattern, line):
+                        print(pattern)
+                        print(line,)
+                        print('Group 1: %s' % (match.group(1)))
+                        print('Group 2: %s' % (match.group(2)))
+                        if match.group(1):
+                            code_key = match.group(1).encode("utf-8")
+                            print("code_key", code_key)
+                            if code_key in l10n_properties:
+                                print("FOUND ", code_key)
+                                l10n_properties[code_key] = 1
+                                props_found += 1
+                                # print(l10n_properties[code_key])
+                        if match.group(2):
+                            code_key = match.group(2).encode("utf-8")
+                            print("code_key", code_key)
+                            if code_key in l10n_properties:
+                                print("FOUND ", code_key)
+                                l10n_properties[code_key] = 1
+                                props_found += 1
+                                # print(l10n_properties[code_key])
+                # en_us_locale_files = os.listdir(l10n_code_src_dir)
+
+    # TODO - find all missing strings task and print out line
+    # further searches and tests for larger insight into string use cases and actual unused strings
+    # gather other useful information
+    # get list of = "literal strings in the code";
+    # Iterate back over code with patterns from not found
+    # or unused strings to find literal strings used and where (line).
+
+    # print report
+    # print (l10n_properties.keys())
+
+    for prop in l10n_properties:
+        print(prop,)
+        print(l10n_properties.get(prop))
+
+    props_count = len(l10n_properties)
+    print("props_count", props_count)
+    print("props_found", props_found)
+    props_not_used = props_count - props_found
+    props_coverage_perc = (float(props_found)/props_count) * 100
+    print("props_not_used", props_not_used)
+    print("props_coverage_perc %{0:.2f}".format(props_coverage_perc))
+    # END
```

Comment:
We should sys.exit(1) if there's a non-zero amount of unused string/strings not in locale files.